WalletWasabi/Tor/Socks5/TorSocks5Client.cs

```diff
@@ -68,12 +76,18 @@ public async Task ConnectAsync()
 
 				try
 				{
-					// Cancellation token for ConnectAsync will be available in .NET 5.
-					await TcpClient.ConnectAsync(host, port.Value).ConfigureAwait(false);
+					await TcpClient.ConnectAsync(host, port.Value, cancellationToken).ConfigureAwait(false);
+				}
+				catch (SocketException ex) when (ex.ErrorCode is 10061 or 111 or 61)
+				{
+					// 10061 ~ "No connection could be made because the target machine actively refused it" on Windows.
+					// 111   ~ "Connection refused" on Linux.
+					// 61    ~ "Connection refused" on macOS.
+					throw new TorConnectionException($"Could not connect to Tor SOCKSPort at '{host}:{port}'. Is Tor running?", ex);
 				}
-				catch (Exception ex) when (IsConnectionRefused(ex))
+				catch (Exception ex) when (ex.Message is string && (ex.Message.StartsWith(ExPrefixOnWindows) || ex.Message.StartsWith(ExPrefixOnUnixBasedOSs)))
```

Comment:
This is not okay. Never compare exception messages because they depend on the user culture. Also, I think those texts correspond to the SocketException ErrorCode that are already controlled above.

Reply:
I would actually love to remove those exception messages. If you guys agree, I can do it.

Reply:
please do it.

Reply:
Done.

Reply:
If that is removable that is the best. Just one comment regarding the language. We are always getting messages in English because of this setting, which was added exactly for this purpose. 

https://github.com/zkSNACKs/WalletWasabi/blob/4c4142d2223db96911af64d4e7f108043a3dd04d/WalletWasabi/WalletWasabi.csproj#L81